exit.go

```diff
@@ -4,127 +4,165 @@ import (
 	"strconv"
 
 	"github.com/cyverse-de/dockerops"
-	"github.com/cyverse-de/logcabin"
 	"github.com/cyverse-de/messaging"
 	"github.com/cyverse-de/model"
+	"github.com/pkg/errors"
 )
 
-func cleanup(job *model.Job) {
-	logcabin.Info.Printf("Performing aggressive clean up routine...")
-
-	logcabin.Info.Println("Finding all input containers")
-	inputContainers, err := dckr.ContainersWithLabel(dockerops.TypeLabel, strconv.Itoa(dockerops.InputContainer), true)
+// RemoveVolume removes the docker volume with the provided volume identifier.
+func RemoveVolume(id string) error {
+	var (
+		err       error
+		hasVolume bool
+	)
+	hasVolume, err = dckr.VolumeExists(id)
 	if err != nil {
-		logcabin.Error.Print(err)
-		inputContainers = []string{}
+		return errors.Wrap(err, "failed to check for volume existence")
 	}
-	for _, ic := range inputContainers {
-		logcabin.Info.Printf("Nuking input container %s", ic)
-		err = dckr.NukeContainer(ic)
-		if err != nil {
-			logcabin.Error.Print(err)
+	if hasVolume {
+		log.Infof("removing volume: %s", id)
+		if err = dckr.RemoveVolume(id); err != nil {
+			return errors.Wrap(err, "failed to remove volume")
 		}
 	}
+	return nil
+}
 
-	logcabin.Info.Println("Finding all step containers")
-	stepContainers, err := dckr.ContainersWithLabel(dockerops.TypeLabel, strconv.Itoa(dockerops.StepContainer), true)
+// RemoveJobContainers removes containers based on their job label.
+func RemoveJobContainers(id string) error {
+	log.Infof("Finding all containers with the label %s=%s", model.DockerLabelKey, id)
+	jobContainers, err := dckr.ContainersWithLabel(model.DockerLabelKey, id, true)
 	if err != nil {
-		logcabin.Error.Print(err)
+		return errors.Wrapf(err, "failed to find containers with label %s=%s", model.DockerLabelKey, id)
 	}
-	for _, sc := range stepContainers {
-		logcabin.Info.Printf("Nuking step container %s", sc)
-		err = dckr.NukeContainer(sc)
+	for _, jc := range jobContainers {
+		log.Infof("Nuking container %s", jc)
+		err = dckr.NukeContainer(jc)
 		if err != nil {
-			logcabin.Error.Print(err)
+			return errors.Wrapf(err, "failed to remove container %s", jc)
```

Comment:
I may be missing something here, but simply logging the error and continuing with the loop seems like the right thing to do here. Otherwise, an error that occurs while removing the first container in the list will prevent the rest of the containers from being removed.